pinot-tools/src/main/java/org/apache/pinot/tools/filesystem/PinotFSBenchmarkDriver.java

```diff
@@ -0,0 +1,252 @@
+  private File _localTempDir;
+  private int _numSegmentsForListFilesTest;
+  private int _dataSizeInMBsForCopyTest;
+
+  public PinotFSBenchmarkDriver(String mode, String configFilePath, String baseDirectoryUri, String localTempDir,
+      Integer numSegmentsForListFilesTest, Integer dataSizeInMBsForCopyTest) throws ConfigurationException {
+    Configuration configuration = new PropertiesConfiguration(new File(configFilePath));
+    PinotFSFactory.init(configuration);
+    _mode = mode;
+    _baseDirectoryUri = URI.create(baseDirectoryUri);
+    _pinotFS = PinotFSFactory.create(_baseDirectoryUri.getScheme());
+    _localTempDir =
+        (localTempDir != null) ? new File(localTempDir) : new File(FileUtils.getTempDirectory(), "benchmark");
+    _numSegmentsForListFilesTest =
+        (numSegmentsForListFilesTest != null) ? numSegmentsForListFilesTest : DEFAULT_NUM_SEGMENTS_FOR_LIST_TEST;
+    _dataSizeInMBsForCopyTest =
+        (dataSizeInMBsForCopyTest != null) ? dataSizeInMBsForCopyTest : DEFAULT_DATA_SIZE_IN_MB_FOR_COPY_TEST;
+    LOGGER.info("PinotFS has been initialized sucessfully. (mode = {}, pinotFSClass = {}, configFile = {}, "
+            + "baseDirectoryUri = {}, localTempDir = {}, numSegmentsForListFilesTest = {}, "
+            + "dataSizeInMBsForCopyTest = {})", _mode, _pinotFS.getClass().getSimpleName(), configFilePath,
+        baseDirectoryUri, _localTempDir, _numSegmentsForListFilesTest, _dataSizeInMBsForCopyTest);
+  }
+
+  public void run() throws Exception {
+    prepareBenchmark();
+
+    switch (_mode.toUpperCase()) {
+      case "ALL":
+        testListFilesInMultipleDirectories();
+        testListFiles();
+        testCopies();
+        break;
+      case "LISTFILES":
+        testListFiles();
+        break;
+      case "COPY":
+        testCopies();
+        break;
+      default:
+        throw new RuntimeException("Not Supported Mode: " + _mode);
+    }
+    cleanUpBenchmark();
+  }
+
+  private void prepareBenchmark() throws IOException {
+    // Clean up base directory
+    if (_pinotFS.exists(_baseDirectoryUri)) {
+      _pinotFS.delete(_baseDirectoryUri, true);
+    }
+
+    if (_localTempDir.exists()) {
+      _localTempDir.delete();
+    }
+
+    // Set up the base directory
+    _pinotFS.mkdir(_baseDirectoryUri);
+    _localTempDir.mkdir();
+  }
+
+  private void cleanUpBenchmark() throws IOException {
+    _pinotFS.delete(_baseDirectoryUri, true);
+    FileUtils.deleteQuietly(_localTempDir);
+    LOGGER.info("Working directories have been cleaned up successfully.");
+  }
+
+  private void testListFilesInMultipleDirectories() throws Exception {
+    LOGGER.info("========= List Files in Multiple Directories ==========");
+    long prepareTime = System.currentTimeMillis();
+    URI listTestUri = combinePath(_baseDirectoryUri, "listTestMultipleFile");
+    _pinotFS.mkdir(listTestUri);
+    LOGGER.info("Created {} for list test...", listTestUri);
+
+    int numDirectories = 5;
```

Comment:
I suppose eventually we want to take these as an argument or config?  Otherwise maybe a static final int

Reply:
changed to static final